opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java

```diff
@@ -204,6 +204,42 @@ public SELF open_banking_reads_anton_brueckner_transactions_using_consent_bound_
         return self();
     }
 
+    @SneakyThrows
+    public SELF open_banking_reads_user_transactions_using_consent_bound_to_service_session_data_validated_by_iban(
+            String user, String resourceId, LocalDate dateFrom, LocalDate dateTo, String bookingStatus
+    ) {
+        ExtractableResponse<Response> response = getTransactionListFor(user, resourceId, dateFrom, dateTo, bookingStatus);
+
+        this.responseContent = response.body().asString();
+        DocumentContext body = JsonPath.parse(responseContent);
+
+        assertThat(body).extracting(it -> it.read("$.transactions.booked[*].creditorAccount.iban")).asList()
+                .containsOnly(
+                        "DE97500105179747157812"
```

Comment:
Creating random user with the same IBAN as a possibly existing user (like test ran twice so two user attempt to exist with the same IBAN) - won't it cause constraint violations on Sandbox side? (as IBAN should be unique)

Reply:
random IBAN generation added